### PR TITLE
#d2l add 8.4 chapter rnn.md

### PR DESCRIPTION
**Pull request by Shelly111111**

### CODE REVIEW

**Comment by XYZ-916** [comment]
10.1review无问题

**Comment by w5688414** [comment]
8.4.把这两部分代码放到一个cell里面：

image

**Comment by 1649759610** [comment]
同学你好，@Shelly111111 ，以下问题请继续检查和修改一下。

### 9.7节
Seq2SeqEncoder 中：

#@save
class Seq2SeqEncoder(d2l.Encoder):
    """用于序列到序列学习的循环神经网络编码器"""
    def __init__(self, vocab_size, embed_size, num_hiddens, num_layers,
                 dropout=0, **kwargs):
        super(Seq2SeqEncoder, self).__init__(**kwargs)
        weight_ih_attr = paddle.ParamAttr(initializer=nn.initializer.XavierUniform())
        weight_hh_attr = paddle.ParamAttr(initializer=nn.initializer.XavierUniform())
        # 嵌入层
        self.embedding = nn.Embedding(vocab_size, embed_size)
        self.rnn = nn.GRU(embed_size, num_hiddens, num_layers,
                          dropout=dropout, weight_ih_attr=weight_ih_attr, weight_hh_attr=weight_hh_attr)

    def forward(self, X, *args):
        # 输出'X'的形状：(batch_size,num_steps,embed_size)
        X = self.embedding(X)
        # 在循环神经网络模型中，第一个轴对应于时间步
        # 如果未提及状态，则默认为0
        output, state = self.rnn(X)
        output = output.transpose([1, 0, 2])
        # PaddlePaddle的GRU层output的形状:(batch_size,time_steps,num_directions * hidden_size)
        # 需将其转置为(num_steps,batch_size,num_directions * num_hiddens)
        # state[0]的形状:(num_layers,batch_size,num_hiddens)
        return output, state


Paddle和torch的默认输入shape形式不一致，所以上述代码的逻辑上是OK的。但为了和原书进行友好的对应，方便读者进行对照阅读，建议在实例化GRU的时候，直接设置time_major=True，这样forward的代码逻辑可以和原书进行友好的对应。
请全节check一下这个问题。另外，如下注释对不齐:num_directions * hidden_size —>num_directions * num_hiddens

Seq2SeqDecoder中：        
X_and_context = d2l.concat((X, context), 2)    —>         X_and_context = paddle.concat((X, context), 2)
X = d2l.ones([2, 3, 4])  —> X = paddle.ones([2, 3, 4])
类似的问题，请全节排查一遍

**Comment by 1649759610** [comment]
9.7节 review 无问题

**Comment by ZhangHandi** [comment]
13.7 LGTM

File: Dive-into-DL-paddlepaddle/docs/13_computer-vision/fcn.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+```
+
+## [**预测**]
+
+在预测时，我们需要将输入图像在各个通道做标准化，并转成卷积神经网络所需要的四维输入格式。
+
+
+```{.python .input}
+#@tab paddlepaddle
+def predict(img):
```
Comment (ZhangHandi) [inline Dive-into-DL-paddlepaddle/docs/13_computer-vision/fcn.md:187]: def predict(img):
    X = test_iter.dataset.normalize_image(img).unsqueeze(0)
    pred = net(X).argmax(axis=1)
    return pred.reshape([pred.shape[1], pred.shape[2]])

这段代码改成这样吧，和书中保持一致

File: Dive-into-DL-paddlepaddle/docs/13_computer-vision/fcn.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    pred = net(X).argmax(axis=1)
+    return pred.reshape([pred.shape[1], pred.shape[2]])
+```
+
+为了[**可视化预测的类别**]给每个像素，我们将预测类别映射回它们在数据集中的标注颜色。
+
+
+```{.python .input}
+#@tab paddlepaddle
+import numpy as np
```
Comment (ZhangHandi) [inline Dive-into-DL-paddlepaddle/docs/13_computer-vision/fcn.md:198]: def label2image(pred):
    colormap = paddle.to_tensor(d2l.VOC_COLORMAP)
    X = paddle.to_tensor(pred, dtype='int64')
    return colormap[X, :]

同理，和书中保持一致

**Comment by ZhangHandi** [comment]
13.11 LGTM

**Comment by ZhangHandi** [comment]
13.13 13.14 LGTM

**Comment by 1649759610** [comment]
15.1 review 无问题

**Comment by tngt** [comment]
8.4、8.7、9.7、10.1、12.4、13.7、13.11、13.13、13.14、15.1review通过
